static/js/browser2/statvar_hierarchy.tsx

```diff
@@ -22,51 +22,76 @@
 import React from "react";
 import axios from "axios";
 import _ from "lodash";
+
 import { StatVarHierarchySearch } from "./statvar_hierarchy_search";
 import { StatVarGroupNode } from "./statvar_group_node";
 import { StatVarGroupNodeType, StatVarNodeType } from "./types";
-import { loadSpinner, removeSpinner } from "./util";
 import { NamedPlace } from "../shared/types";
 
+import { loadSpinner, removeSpinner } from "./util";
+import { Context } from "../shared/context";
+
 const LOADING_CONTAINER_ID = "stat-var-hierarchy-section";
 const SORTED_FIRST_SVG_ID = "dc/g/Demographics";
 const SORTED_LAST_SVG_ID = "dc/g/Miscellaneous";
 
 interface StatVarHierarchyPropType {
+  type: string;
   places: NamedPlace[];
+  // (Optional) A list of stat vars selected from parent componenet.
+  // For example, in timeline tool, these are stat vars parsed from URL.
+  svs?: string[];
 }
 
 interface StatVarHierarchyStateType {
-  statVarGroups: { [svgId: string]: StatVarGroupNodeType };
-  statVars: { [statVarId: string]: StatVarNodeType };
-  pathToSelection: string[];
+  focus?: string;
+  // The path to the focus node. If the focus is an svg, it will not be
+  // stored in this.state.svPath.
+  focusPath: string[];
+  // A map from stat var dcid to its path in the hierarchy.
+  svPath: Record<string, string[]>;
+  // Error message.
   errorMessage: string;
+  // Select or de-select a stat var with its path.
+  togglePath: (sv: string, path?: string[]) => void;
 }
 
 export class StatVarHierarchy extends React.Component<
   StatVarHierarchyPropType,
   StatVarHierarchyStateType
 > {
+  // A map from stat var group dcid to the metadata.
+  svgInfo: { [svgId: string]: StatVarGroupNodeType };
+  // A map from stat var dcid to the metadata.
+  svInfo: { [svId: string]: StatVarNodeType };
+  // The stat var (group) from search result. The hierarchy focuses on it by
+  // displaying the path from root to it.
+
```

Comment:
is there a missing variable here for the sv group?

Reply:
It's the svgInfo on line 64